machine-runner/README.md

```diff
@@ -225,6 +225,63 @@ Instead, each robot independently ensures that after at five seconds a decision
 Machine runner resolves this conflict by using only the `selected` event that comes first in the Actyx event sort order; in other words, Actyx arbitrarily picks a winner and the losing event is discarded.
 If a robot saw itself winning, started the mission, and then discovers that its win turned out to be invalid, it will have to stop the mission and pick a new one.
 
+## Other Features
+
+### Extra Tags
+
+In the case extra tags are required to be attached in events when invoking commands, these following APIs can be used.
+
+#### On Command Definition
+
+Extra tags can be registered on a command definition. These extra tags will always be attached when the command is invoked.
+
+```typescript
+// State definition
+export const InitialWarehouse = TransportOrderForWarehouse.designState('Initial')
+  .withPayload<{ id: string }>()
+  .command('request', [requested], (ctx, from: string, to: string) => {
+    ctx.withTags([
+      `transport-order-from:${from}`,
+      `transport-order-to:${to}`
+    ])
```

Comment:
I'm partial toward a single list argument instead of relying on an argument list because:
- the array "square" syntax is portable when editing. It's easy to take in and take out.
- it doesn't close the function signature from additional arguments with different types when needed.

What do you think?

Reply:
The name `.withTags` is very specific, so it is unlikely to ever take more arguments. It also includes the plural form “tags” which makes it intuitive to supply multiple of them as arguments. “Portable when editing” is not fully clear to me: the formatting is just about white space and should be customisable using prettierrc or similar, right? One thing I’d like to avoid is that this function always requires multiple lines, even when the contents would fit nicely on a single line — that feels a bit like bullying to me (I know that single-line-arrays can be configured in prettier but I don’t recall the default setting).

Reply:
On "Portable when editing", might be a misnomer, but what I meant by this is that pulling out a list of items into a variable is easier. For example:

```
// with argument list
// turning:
withTags("a", "b", "c")
// into:
const tags = ["a", "b", "c"]
withTags(...tags)

// requires these steps
1. cut the content: `withTags()  --cut--> "a", "b", "c"`
2. write var and create a bracket: `const tags = []`  
3. paste content:  `const tags = [<--paste-- "a", "b", "c"]`
4. add `...tags` into the param: `withTags(...tags)`
```
meanwhile
```
// with single argument
// turning
withTags(["a","b","c"])
// into
const tags = ["a", "b", "c"]
withTags(tags)

// requires these steps:
1. cut the content: `withTags()  --cut--> ["a", "b", "c"]`
2. write var and paste the value: `const tags = <--paste-- ["a", "b", "c"]`
3. add `tags` into the param: `withTags(tags)`
```

The first signature requires more precision for "putting the cursor right" before cutting and pasting. 
The `three dots` of the param also requires counting to three at the back of one's head.
And on step 3, if done without the bracket, would invoke a slight feeling of being punished because the editor would bleep a red marker for a bit `const tags = "a", "b", "c"` is not a valid TS, JS syntax

This seems simple, but the reason I feel strongly about this is because I did quite a lot of refactoring that involves an argument list and to me, it has a higher cognitive cost.

But I'm not sure how other people would perceive this.

Reply:
Thanks for the explanation! As I said, as long as our API allows single-line use in standard prettier/eslint configurations I’m fine either way.